acsoo/tools.py

```diff
@@ -26,6 +28,55 @@ def _escape(s):
     return s
 
 
+@contextmanager
+def tempinput(data):
+    temp = tempfile.NamedTemporaryFile(delete=False)
+    temp.write(data)
+    temp.close()
+    try:
+        yield temp.name
+    finally:
+        os.unlink(temp.name)
+
+
+def cmd_check_current_branch(branches):
+    current_branch = check_output(
+        ['git', 'rev-parse', '--abbrev-ref', 'HEAD'])
+    current_branch = current_branch.replace("\n", "")
+    if current_branch not in branches:
+        return False
+    else:
+        return True
+
+
+def cmd_push(paths_to_push, message, skip_ci=True, user_name='',
+             user_email=''):
+    out = check_output(['git', 'ls-files', '--other', '--exclude-standard',
+                        '--modified'])
+    list_out = out.splitlines()
+    items_to_remove = set([])
+    for item in paths_to_push:
+        if item not in list_out:
+            click.echo("No change on %s" % item)
+            items_to_remove.add(item)
+    to_push = set(paths_to_push) - items_to_remove
+    if to_push:
+        click.echo(out)
+        add_cmd = ['git', 'add']
+        add_cmd.extend(paths_to_push)
+        check_call(add_cmd)
+        if skip_ci:
+            message = "%s [ci skip]" % message
+        if user_name:
+            check_call(['git', 'config', 'user.name', user_name])
+        if user_email:
+            check_call(['git', 'config', 'user.email', user_email])
```

Comment:
hm, in the shell executor these global git config are not ideal.

Even on the developer's PC, I find it surprising to have acsoo changing my git config.

Why do you need this?

Reply:
Only for testing with gitlab :smile: I'll propose a change in the configuration deployement tool to add identities on gitlab-runner.